src/actions.js

```diff
@@ -0,0 +1,48 @@
+const add = (tasks, input) => {
+  const task = {
+    description: input,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(task);
+  return tasks;
+};
+
+const shiftIndex = (tasks, taskIndex) => {
+  tasks = tasks.map((task) => {
+    if (task.index > taskIndex) {
+      task.index -= 1;
+    }
+    return task;
+  });
+  return tasks;
+};
+
+const remove = (tasks, task) => {
+  const taskIndex = tasks.findIndex((item) => (item.index === task.index));
+  tasks = tasks.filter((item) => (item.index !== task.index));
+  tasks = shiftIndex(tasks, taskIndex);
+  return tasks;
+};
+
+const edit = (tasks, task, desc) => {
```

Comment:
**OPTIONAL**

- I highly suggest that you manage to edit the task's value directly without the use of a form to do that. This would enhance the user experience of your app. 😄 🙏

Reply:
Done. Now the user can edit the task's value directly